File: C# .Net/JDI UI Framework/JDI/Core/Interfaces/Complex/IDropDown.cs
using System;
using Epam.JDI.Core.Interfaces.Base;
using Epam.JDI.Core.Interfaces.Common;

namespace Epam.JDI.Core.Interfaces.Complex
{
    public interface IDropDown : IDropDown<IConvertible> { }
    public interface IDropDown<in TEnum> : ISelector<TEnum>, IText, IClickable
        where TEnum : IConvertible
    {
        /**
     * Expanding DropDown
     */
        //TODO[JDIAction]
        void Expand();

        /**
         * Closing DropDown
         */
        //TODO[JDIAction]
        void Close();
    }
}

File: Java/JDI/jdi-uitest-core/src/main/java/com/epam/jdi/uitests/core/interfaces/complex/IDropDown.java
package com.epam.jdi.uitests.core.interfaces.complex;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

import com.epam.jdi.uitests.core.annotations.JDIAction;
import com.epam.jdi.uitests.core.interfaces.base.IClickable;
import com.epam.jdi.uitests.core.interfaces.common.IText;

/**
 * Created by Roman_Iovlev on 6/10/2015.
 */
public interface IDropDown<TEnum extends Enum> extends ISelector<TEnum>, IText, IClickable {
    /**
     * Expanding DropDown
     */
    @JDIAction
    void expand();

    /**
     * Closing DropDown
     */
    @JDIAction
    void close();

}